executor/include/execution/trait/can_prefer.hpp

```diff
@@ -0,0 +1,34 @@
+//
+// Created by Shrijit Singh on 17/06/20.
+//
+
+#pragma once
+
+#include <execution/trait/can_require.hpp>
+#include <type_traits>
+
+namespace execution {
+
+template <typename Executor, typename Property,
+          typename std::enable_if_t<
+              Property::template is_applicable_property_v<Executor> &&
+                  Property::is_preferable,
+              int> = 0>
+Executor prefer(Executor ex, const Property t) {
+  if (execution::can_require_v<Executor, Property>) return ex.require(t);
+  return ex;
+}
+
+template <typename Executor, typename Properties, typename = void>
+struct can_prefer : std::false_type {};
+
+template <typename Executor, typename Property>
+struct can_prefer<Executor, Property,
+                  std::void_t<decltype(execution::require(
+                      std::declval<Executor>(), std::declval<Property>()))>>
```

Comment:
Is `can_prefer` and `can_require` the same?

Reply:
No, it seems like a bug.
Fixed in 089ef0421296e37ee29b72e8ffb3383bd6ab0fdc